primap2/tests/test_setters.py

```diff
@@ -0,0 +1,383 @@
+"""Tests for _setters.py"""
```

Comment:
I have to learn pytest before I can say much about the tests

Reply:
the most interesting concepts are fixtures and pytest.mark.parametrize, the rest of pytest is pretty straight-forward, I think.